### PR TITLE
Ubuntu dod repo init obs

### PR DESCRIPTION
**Pull request by reSHARMA**
Since @athos-ribeiro has already covered sid, jessie and stretch (many thanks for covering jessie and stretch). 

I've added for ubuntu, and have kept it in a different state file to isolate it from installation (though it can be considered as a part of installation)

### CODE REVIEW

File: llvm-obs/dod-repo-conf/artful.conf [hunk 1]
```diff
@@ -0,0 +1,249 @@
+
+Repotype: debian
+
+Preinstall: bash perl-base libperl5.26 perl-modules-5.26 sed grep coreutils debianutils
```
Comment (sylvestre) [inline llvm-obs/dod-repo-conf/artful.conf:4]: I don't think it is what we want.
It won't scale as it will break when perl or other packages will be renamed.

File: llvm-obs/dod-repo-conf/artful.conf [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+Runscripts: base-files base-passwd passwd sysv-rc
+
+VMinstall: binutils libblkid1 libuuid1 mount libmount1 libsmartcols1
+%ifarch ppc64le
+# for LE switch
+VMinstall: kernel-obs-build
+%endif
+
+# Experimental: testing OBS snapcraft support, adrian
```
Comment (sylvestre) [inline llvm-obs/dod-repo-conf/artful.conf:32]: I would prefer avoiding snap

File: llvm-obs/dod-repo-conf/artful.conf [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+Support: make man
+Support: net-tools
+Support: patch procps psmisc strace
+Support: unzip vim ncurses-base
+Support: locales
+Support: mount
+#Support: module-init-tools texinfo
+
+# circular dependendencies in openjdk stack
+Order: openjdk-6-jre-lib:openjdk-6-jre-headless
```
Comment (sylvestre) [inline llvm-obs/dod-repo-conf/artful.conf:72]: why is that needed?

File: llvm-obs/dod-repo-conf/artful.conf [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+Prefer: -libgcc-mainline -libstdc++-mainline -gcc-mainline-c++
+Prefer: -libgcj-mainline -viewperf -compat -compat-openssl097g
+Prefer: -zmd -OpenOffice_org -pam-laus -libgcc-tree-ssa -busybox-links
+Prefer: -crossover-office -libjack-dev
+Prefer: -libjpeg-turbo8-dev -libblas3gf
+
+Conflict: ghostscript-library:ghostscript-mini
+
+Ignore: libgtk2.0-0:gnome-icon-theme
```
Comment (sylvestre) [inline llvm-obs/dod-repo-conf/artful.conf:112]: what is this ignore list?

File: llvm-obs/dod-repo-conf/artful.conf [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+Substitute: java2-devel-packages java-1_4_2-sun-devel
+%else
+ %ifnarch s390x
+Substitute: java2-devel-packages java-1_4_2-ibm-devel
+ %else
+Substitute: java2-devel-packages java-1_4_2-ibm-devel xorg-x11-libs-32bit
+ %endif
+%endif
+
+%ifarch %ix86
```
Comment (sylvestre) [inline llvm-obs/dod-repo-conf/artful.conf:222]: please comment the unused archs

File: llvm-obs/dod-repo-conf/artful.conf [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+Ignore: libcdio:libcddb.so.2
+Ignore: gnome-libs:libgnomeui
+Ignore: nautilus:gnome-themes
+Ignore: gnome-panel:gnome-themes
+Ignore: gnome-panel:tomboy
+
+Substitute: utempter
+
+%ifnarch s390 s390x ppc ia64
+Substitute: java2-devel-packages java-1_4_2-sun-devel
```
Comment (sylvestre) [inline llvm-obs/dod-repo-conf/artful.conf:213]: these packages are from the past :)

File: llvm-obs/obs-scripts/dod-ubuntu-init.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# Generates dod repos for all supported Ubuntu versions in obs
```
Comment (sylvestre) [inline llvm-obs/obs-scripts/dod-ubuntu-init.sh:2]: A bunch of them have been deprecated already

File: llvm-obs/dod-repo-conf/artful.conf (hunk 1 above)
Comment (reSHARMA) [reply llvm-obs/dod-repo-conf/artful.conf:4]: We were using these configurations directly from openSUSE obs reference server.
for eg https://build.opensuse.org/project/prjconf/Debian:9.0

Do we need to generate this by our own?

File: llvm-obs/obs-scripts/dod-ubuntu-init.sh (hunk 7 above)
Comment (reSHARMA) [reply llvm-obs/obs-scripts/dod-ubuntu-init.sh:2]: But snapshot branch has build job for all these in the Jenkins instance? 
And for branch 6.0 we don't build for all these. Are you referring to that? :)
